env/production/aws-iam-role-GitHubActionsRoleNextstrainBatchJobs.tf

```diff
@@ -22,10 +22,16 @@ resource "aws_iam_role" "GitHubActionsRoleNextstrainBatchJobs" {
         "Condition": {
           "StringLike": {
             "token.actions.githubusercontent.com:aud": "sts.amazonaws.com",
-            "token.actions.githubusercontent.com:sub": [
-              for repo in keys(local.repo_pathogens):
-                "repo:nextstrain/${repo}:*:job_workflow_ref:nextstrain/.github/.github/workflows/pathogen-repo-build.yaml@*"
-            ]
+            "token.actions.githubusercontent.com:sub": flatten([
+              # Special case for seasonal flu's deploy-private-nextflu workflow which needs to download the private builds
+              # from AWS Batch before bundling/deploying them through Netlify.
+              # This special case can be removed when we finally sunset the private site.
+              #   -Jover, 07 June 2024
+              "repo:nextstrain/seasonal-flu:*:job_workflow_ref:nextstrain/seasonal-flu/.github/workflows/deploy-private-nextflu.yaml@*",
```

Comment:
Ah, just reading the [OIDC docs](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect#understanding-the-oidc-token) more closely, `job_workflow_ref` is for called _reusable_ workflows only...what we need here would be `workflow_ref`

Reply:
Oy. Really wishing GitHub allowed mapping of other claims in its token so we could use PrincipalTags here without a bunch of effort. But they don't, so.

We could add `workflow_ref` to the `sub` claim and adjust other conditions to account for it too, e.g. 

    …/pathogen-repo-build@*:workflow_ref:*

or change nothing about `sub` and simply let everything in seasonal-flu access Batch for now.

    repo:nextstrain/seasonal-flu:*

Reply:
The first option would look like 3dd92ddcb0d9402333c072ebec317ca68d3ea0c0.

Reply:
> The first option would look like https://github.com/nextstrain/infra/commit/3dd92ddcb0d9402333c072ebec317ca68d3ea0c0.

Thanks! I'll cherry-pick and deploy the changes to give it a try.

Reply:
Here's hoping it works…

Reply:
And of course [it doesn't work](https://github.com/nextstrain/seasonal-flu/actions/runs/9424047397)...

Reply:
Rolled back the Terraform changes for now.

Reply:
:man_facepalming: sorry

Reply:
Trying the second option with https://github.com/nextstrain/infra/pull/19/commits/81b8676bc775ab33f415e17f7ce581eba2aa0e63

Reply:
...and [that's working as expected](https://github.com/nextstrain/seasonal-flu/actions/runs/9424047397) 🎉

Reply:
\o/ Kinda surprised that adding the missing `@*` wasn't enough... wonder what it was.

Reply:
I'm leaving that mystery for another time...